File: src/lib/java/work/lclpnet/ap2/impl/game/kit/KitReadView.java
package work.lclpnet.ap2.impl.game.kit;

import net.minecraft.server.level.ServerPlayer;
import org.jetbrains.annotations.NotNull;

public interface KitReadView {

    @NotNull Kit getKit(ServerPlayer player);

    boolean hasKitEquipped(ServerPlayer player, Kit kit);
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/kit/KitReadView.kt
package work.lclpnet.ap2.game.kit

import net.minecraft.server.level.ServerPlayer

interface KitReadView {
    fun getKit(player: ServerPlayer): Kit

    fun hasKitEquipped(player: ServerPlayer, kit: Kit): Boolean =
        getKit(player) == kit
}

inline fun <reified T> KitReadView.hasKitEquipped(player: ServerPlayer): Boolean =
    getKit(player) is T